sidecar/nginx/lua/amalgam8.lua

```diff
@@ -691,6 +686,18 @@ function Amalgam8:apply_rules()
    local selected_backend = nil
    local cookie_version = ngx.var.cookie_version --check for version cookie
 
+   -- if we have routes/actions for the destination but no instances, then return HTTP 503
+   -- else return HTTP 404, as we don't know if this service actually exists or not.
+
+   if not instances or #instances == 0 then
+      if routes or actions then
+         ngx.status = ngx.HTTP_SERVICE_UNAVAILABLE
```

Comment:
I follow that you want to distinguish between no instances found vs no service found - but do we really want to through a 503 error code?  Aren't 5xx codes typically server faults?  If the user hasn't created any instances for that service - is it the proxy's fault?